packages/babel-helper-replace-supers/src/index.js

```diff
@@ -188,139 +176,143 @@ export default class ReplaceSupers {
-    const node = path.node;
-
-    if (isIllegalBareSuper(node, parent)) {
-      throw path.buildCodeFrameError("Illegal use of bare super");
+  specHandleAssignmentExpression(path) {
+    const { node } = path;
+    const { operator } = node;
+    const { computed, property } = node.left;
+    if (operator === "=") {
+      // super.name = "val"
+      // to
+      // _set(Object.getPrototypeOf(objectRef.prototype), "name", this);
+      const setter = this.setSuperProperty(property, node.right, computed);
+      return [setter];
     }
 
-    if (t.isCallExpression(node)) {
-      const callee = node.callee;
-      if (t.isSuper(callee)) {
-        return;
-      } else if (isMemberExpressionSuper(callee)) {
-        // super.test();
-        // to
-        // _get(Object.getPrototypeOf(objectRef.prototype), "test", this).call(this);
-        property = callee.property;
-        computed = callee.computed;
-        args = node.arguments;
-      }
-    } else if (t.isMemberExpression(node) && t.isSuper(node.object)) {
-      // super.name;
-      // to
-      // _get(Object.getPrototypeOf(objectRef.prototype), "name", this);
-      property = node.property;
-      computed = node.computed;
-    } else if (
-      t.isUpdateExpression(node) &&
-      isMemberExpressionSuper(node.argument)
-    ) {
-      const binary = t.assignmentExpression(
-        node.operator[0] + "=",
-        node.argument,
+    // super.age += 2;
+    // to
+    // _set(
+    //   Object.getPrototypeOf(objectRef.prototype),
+    //   "name",
+    //   _get(Object.getPrototypeOf(CLASS.prototype), "METHOD", this) + 2,
+    //   this,
+    //   true,
+    // );
+    // TODO this needs cleanup. Should be a single proto lookup
+    const ref = path.scope.generateDeclaredUidIdentifier("ref");
+    const setter = this.setSuperProperty(
+      property,
+      t.binaryExpression(operator.slice(0, -1), t.cloneNode(ref), node.right),
+      computed,
+    );
+    return [
+      t.assignmentExpression(
+        "=",
+        t.cloneNode(ref),
+        this.getSuperProperty(property, computed),
+      ),
+      setter,
+    ];
+  }
+
+  specHandle(path: NodePath) {
+    const { node, parentPath } = path;
+    const grandParentPath = parentPath.parentPath;
+    let parent = parentPath.node;
+
+    if (grandParentPath.isUpdateExpression({ argument: parent })) {
+      const { operator, prefix } = grandParentPath.node;
+      const assignment = t.assignmentExpression(
+        operator[0] + "=",
+        parent,
         t.numericLiteral(1),
       );
-      if (node.prefix) {
+      grandParentPath.replaceWith(assignment);
+
+      let nodes;
+
+      // TODO this needs some more cleanup.
+      if (prefix) {
         // ++super.foo;
         // to
-        // super.foo += 1;
-        return this.specHandleAssignmentExpression(null, path, binary);
+        // _ref = super.foo, super.foo = _ref + 1
+        nodes = this.specHandleAssignmentExpression(grandParentPath);
```

Comment:
I think that we should convert `super.foo` to a number, like we do for `super.foo++`?

Reply:
Yup, good catch.